src/app/components/login-page/login-page.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, OnInit } from '@angular/core';
+import { LoginService } from '../../services/login-service/login-service.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-login-page',
+  templateUrl: './login-page.component.html',
+  styleUrls: ['./login-page.component.scss']
+})
+export class LoginPageComponent implements OnInit {
+
+  usernameInput: string;
+  passwordInput: string;
+  loginMessage: string;
+
+  constructor(private loginService: LoginService, public router: Router) { }
+
+  sendLogin() {
+    if (this.usernameInput && this.passwordInput) {
+      this.loginService.login(this.usernameInput, this.passwordInput).subscribe((response) => {
+        if (response.statusCode === 200) {
+          this.router.navigate([`/heroes-hall/${response.id}`]);
+        }
+      },
+      error => {
+          const acceptedResponses = [
+            'Incorrect Username and/or Password!',
+            'Please enter a Username and a Password!'
+          ];
+          if (acceptedResponses.indexOf(error.error.message) > -1) {
```

Comment:
the field nesting (error.error.message) is also only necessary because you dont make your own error response

Reply:
Fixed, thanks, moved error handling logic into the service.